### ISSUE TITLE
PyTorch implementation

### ISSUE THREAD

**Issue by cg563** [anishathalye/obfuscated-gradients#17]
I have been trying to re-implement your attack in PyTorch, but the result I got is very confusing. Using the PyTorch trained Inception-v3 model, the BPDA attack could not achieve a reasonable success rate even against the JPEG transformation defense. I checked that the code is virtually identical to your released Tensorflow code, which certainly does work.

I am suspecting that the PyTorch trained models use a certain pre-processing or augmentation that makes them more robust against adversarial perturbations. I am curious to see if you can reach a similar conclusion.

**Comment by cg563** [comment]
After some tweaking around with hyperparameters, I found that on PyTorch models, it requires more iterations with a smaller learning rate. I'm not sure what the reason is, but the attack can get to a reasonable success rate now. I will close the issue.
